File: Examples/CSharp/GroupDocs.Conversion.Examples.CSharp/ConvertToImage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using GroupDocs.Conversion.Config;
using GroupDocs.Conversion.Converter.Option;
using GroupDocs.Conversion.Handler;

namespace GroupDocs.Conversion.Examples.CSharp
{
    class ConvertToImage
    {
        // Default Sample File Path and output path and GUID File as Input
        public static string storagePath = Path.Combine(Environment.CurrentDirectory, @"..\..\Data\SampleFiles");
        public static string cachePath = Path.Combine(Environment.CurrentDirectory, @"..\..\Data\OutputFiles");
        public static string inputGUIDFile = "DOCXsample.docx";
        //public static string inputGUIDFile = "PDFsample.pdf";

        // Result as file path
        public static void ToImageAsPath()
        {
            // Setup Conversion configuration
            var conversionConfig = new ConversionConfig { StoragePath = storagePath, CachePath = cachePath };

            //instantiating the conversion handler
            var conversionHandler = new ConversionHandler(conversionConfig);

            var convertedDocumentPath = conversionHandler.Convert<IList<string>>(inputGUIDFile, new ImageSaveOptions { ConvertFileType = ImageSaveOptions.ImageFileType.Jpg });
        }

        // Result as Stream
        public static void ToImageAsStream()
        {
            // Setup Conversion configuration
            var conversionConfig = new ConversionConfig { StoragePath = storagePath, CachePath = cachePath };

            //instantiating the conversion handler
            var conversionHandler = new ConversionHandler(conversionConfig);


            var convertedDocumentStream = conversionHandler.Convert<IList<Stream>>(inputGUIDFile, new ImageSaveOptions { ConvertFileType = ImageSaveOptions.ImageFileType.Jpg });
        }

        // Advanced example
        public static void ToImageAdvance()
        {
            // Setup Conversion configuration
            var conversionConfig = new ConversionConfig { StoragePath = storagePath, CachePath = cachePath };
            conversionConfig.SetUseCache(true);

            //instantiating the conversion handler
            var conversionHandler = new ConversionHandler(conversionConfig);

            //Set password to unprotect protected document during loading
            LoadOptions loadOptions = new LoadOptions { Password = "secret" };

            // convert file to Tiff, starting from page 2 and convert 2 pages,
            // use DPI 300, image width 1024, image height 768
            SaveOptions saveOptions = new ImageSaveOptions
            {
                ConvertFileType = ImageSaveOptions.ImageFileType.Tiff,
                PageNumber = 2,
                NumPagesToConvert = 2,
                Dpi = 300,
                Width = 1024,
                Height = 768
            };

            var convertedDocumentStream = conversionHandler.Convert<IList<Stream>>(inputGUIDFile, loadOptions, saveOptions);
        }
    }
}

File: Examples/VisualBasic/GroupDocs.Conversion.Examples.VisualBasic/ConvertToImage.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.IO
Imports GroupDocs.Conversion.Config
Imports GroupDocs.Conversion.Converter.Option
Imports GroupDocs.Conversion.Handler

namespace GroupDocs.Conversion.Examples.VisualBasic
    Class ConvertToImage
        ' Default Sample File Path and output path and GUID File as Input
        Public Shared storagePath As String = Path.Combine(Environment.CurrentDirectory, "..\..\Data\SampleFiles")
        Public Shared cachePath As String = Path.Combine(Environment.CurrentDirectory, "..\..\Data\OutputFiles")
        Public Shared inputGUIDFile As String = "DOCXsample.docx"
        'Public static string inputGUIDFile = "PDFsample.pdf"

        ' Result as file path
        Public Shared Sub ToImageAsPath()
            ' Setup Conversion configuration
			Dim conversionConfig = New ConversionConfig() With { _
				Key .StoragePath = storagePath, _
				Key .CachePath = cachePath _
			}

            'instantiating the conversion handler
            Dim conversionHandler = New ConversionHandler(conversionConfig)

			Dim convertedDocumentPath = conversionHandler.Convert(Of IList(Of String))(inputGUIDFile, New ImageSaveOptions() With { _
				Key .ConvertFileType = ImageSaveOptions.ImageFileType.Jpg _
			})
        End Sub

        ' Result as Stream
        Public Shared Sub ToImageAsStream()
            ' Setup Conversion configuration
			Dim conversionConfig = New ConversionConfig() With { _
				Key .StoragePath = storagePath, _
				Key .CachePath = cachePath _
			}

            'instantiating the conversion handler
            Dim conversionHandler = New ConversionHandler(conversionConfig)


			Dim convertedDocumentStream = conversionHandler.Convert(Of IList(Of Stream))(inputGUIDFile, New ImageSaveOptions() With { _
				Key .ConvertFileType = ImageSaveOptions.ImageFileType.Jpg _
			})
        End Sub

        ' Advanced example
        Public Shared Sub ToImageAdvance()
            ' Setup Conversion configuration
			Dim conversionConfig = New ConversionConfig() With { _
				Key .StoragePath = storagePath, _
				Key .CachePath = cachePath _
			}
            conversionConfig.SetUseCache(True)

            'instantiating the conversion handler
            Dim conversionHandler = New ConversionHandler(conversionConfig)

            'Set password to unprotect protected document during loading
			Dim loadOptions As New LoadOptions() With { _
				Key .Password = "secret" _
			}

            ' convert file to Tiff, starting from page 2 and convert 2 pages,
            ' use DPI 300, image width 1024, image height 768
			Dim saveOptions As SaveOptions = New ImageSaveOptions() With { _
				Key .ConvertFileType = ImageSaveOptions.ImageFileType.Tiff, _
				Key .PageNumber = 2, _
				Key .NumPagesToConvert = 2, _
				Key .Dpi = 300, _
				Key .Width = 1024, _
				Key .Height = 768 _
			}

            Dim convertedDocumentStream = conversionHandler.Convert(Of IList(Of Stream))(inputGUIDFile, loadOptions, saveOptions)
        End Sub
    End Class
End Namespace